plugin/src/main/resources/io/jenkins/plugins/analysis/core/model/ReportScanningTool/help-pattern.html

```diff
@@ -2,4 +2,7 @@
     A pattern is defined by an <a href="https://ant.apache.org/manual/Types/fileset.html">Ant Fileset ''includes''</a>
     setting that specifies the report files to scan for issues. Multiple patterns can be separated by space or comma.
     Note that such a pattern is resolved in Jenkins' workspace, so the paths must be relative only.
- </div>
+    <p>
+    If no pattern is specified then the build's console text will be scanned instead.
+    </p>
```

Comment:
This is not correct see comment above about CheckStyle

Reply:
Hmm, yes...
I think that it would be beneficial to tell people this additional information _when it is true_ though.
I'll see if I can figure out a way of making it conditional on `getPattern()` returning an empty string - I think it's possible to have the help generated by groovy instead of simple HTML...

Reply:
I experimented and found that a help-pattern.jelly file isn't passed enough information to dynamically decide what to do.
So instead I extended the existing choose/when/otherwise pattern in the config.jelly to switch between 3 different help files as well as descriptions.
I've checked and this works for e.g. CCM, Brakeman etc as well as GroovyScript - the help, description and validation now document/enforce the same logic.